extensions/arc/src/controller/auth.ts

```diff
@@ -0,0 +1,77 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as request from 'request';
+import * as vscode from 'vscode';
+
+export interface Authentication {
+	applyToRequest(requestOptions: request.Options): Promise<void> | void;
+}
+
+class SslAuth implements Authentication {
+	constructor() { }
+
+	applyToRequest(requestOptions: request.Options): void {
+		requestOptions['agentOptions'] = {
+			rejectUnauthorized: !getIgnoreSslVerificationConfigSetting()
+		};
+	}
+}
+
+export class KerberosAuth extends SslAuth implements Authentication {
+
+	constructor(public kerberosToken: string) {
+		super();
+	}
+
+	applyToRequest(requestOptions: request.Options): void {
+		super.applyToRequest(requestOptions);
+		if (requestOptions && requestOptions.headers) {
+			requestOptions.headers['Authorization'] = `Negotiate ${this.kerberosToken}`;
+		}
+		requestOptions.auth = undefined;
+	}
+}
+
+export class BasicAuth extends SslAuth implements Authentication {
+	constructor(public username: string, public password: string) {
+		super();
+	}
+
+	applyToRequest(requestOptions: request.Options): void {
+		super.applyToRequest(requestOptions);
+		requestOptions.auth = {
+			username: this.username, password: this.password
+		};
+	}
+}
+
+export class OAuthWithSsl extends SslAuth implements Authentication {
+	constructor(public accessToken: string) {
+		super();
+	}
+
+	applyToRequest(requestOptions: request.Options): void {
+		super.applyToRequest(requestOptions);
+		if (requestOptions && requestOptions.headers) {
+			requestOptions.headers['Authorization'] = `Bearer ${this.accessToken}`;
+		}
+		requestOptions.auth = undefined;
+	}
+}
+
+/* Retrieves the current setting for whether to ignore SSL verification errors */
+export function getIgnoreSslVerificationConfigSetting(): boolean {
+	const arcConfigSectionName = 'azureArc';
+	const ignoreSslConfigName = 'ignoreSslVerification';
+
+	try {
+		const config = vscode.workspace.getConfiguration(arcConfigSectionName);
+		return config.get<boolean>(ignoreSslConfigName, true);
```

Comment:
So currently we aren't contributing this config. I believe we'll still want to have this option so make sure you're adding it like we do for BDC : https://github.com/microsoft/azuredatastudio/blob/master/extensions/big-data-cluster/package.json#L144